test/integration/tls/tls_stack.go

```diff
@@ -0,0 +1,908 @@
+package tls
+
+import (
+	"context"
+	"fmt"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/algorithm"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework/k8s"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework/manifest"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"io/ioutil"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	//If you're not able to access below images, try to build them based on the app code under "timeout_app"
+	//directory and push it to any accessible ECR repo and update the below values
+	defaultFrontEndImage = "928111597794.dkr.ecr.us-west-2.amazonaws.com/amazon/tls-e2e:feapp-tlsimage"
+	defaultBackEndImage  = "928111597794.dkr.ecr.us-west-2.amazonaws.com/amazon/tls-e2e:beapp-tlsimage"
+
+	tlsTest          = "tls-e2e"
+	AppContainerPort = 8080
+
+	tlsConnectionError         = "upstream connect error"
+	expectedBackendResponse = "backend"
+)
+
+//FrontEnd -> Will have TLS Validation enabled in both the test cases
+//BackEnd  -> TLS support will be toggled between the test cases to verify that the communication is possible only
+//            when the TLS handshake works between FrontEnd and BackEnd envoys
+
+type TLSStack struct {
+	// service discovery type
+	ServiceDiscoveryType manifest.ServiceDiscoveryType
+
+	// ====== runtime variables ======
+	mesh      *appmesh.Mesh
+	namespace *corev1.Namespace
+
+	FrontEndVN *appmesh.VirtualNode
+	FrontEndDP *appsv1.Deployment
+
+	BackEndVN  *appmesh.VirtualNode
+	BackEndDP  *appsv1.Deployment
+	BackEndSVC *corev1.Service
+
+	BackEndVS *appmesh.VirtualService
+	BackEndVR *appmesh.VirtualRouter
+}
+
+func (s *TLSStack) DeployTLSStack(ctx context.Context, f *framework.Framework) {
+	s.createTLSStackMeshAndNamespace(ctx, f)
+	s.ServiceDiscoveryType = manifest.DNSServiceDiscovery
```

Comment:
Isn't discovery type passed down from `tls_stack_test.go` ? This will override the values passed from the test